source/Engine.cpp

```diff
@@ -1156,12 +1158,27 @@ void Engine::EnterSystem()
 			asteroids.Add(a.Name(), a.Count(), a.Energy());
 	}
 	
-	// Place five seconds worth of fleets. Check for undefined fleets by not
-	// trying to create anything with no government set.
+	// Clear any active weather events
+	activeWeather.clear();
+	// Place five seconds worth of fleets and weather events. Check for
+	// undefined fleets by not trying to create anything with no
+	// government set.
 	for(int i = 0; i < 5; ++i)
+	{
 		for(const System::FleetProbability &fleet : system->Fleets())
 			if(fleet.Get()->GetGovernment() && Random::Int(fleet.Period()) < 60)
 				fleet.Get()->Place(*system, newShips);
+		for(const System::HazardProbability &hazard : system->Hazards())
+			if(Random::Int(hazard.Period()) < 60)
+			{
+				const Hazard *weather = hazard.Get();
+				int hazardLifetime = weather->Length();
```

Comment:
```suggestion
				const int hazardLifetime = weather->Length();
```

Reply:
While this variable (and many others that you've commented about) is "constant," the scope of the variable doesn't leave this if statement (or a single for loop iteration), so labeling it as const seem unnecessary. (See #4520 where a similar scenario occurs where I labeled various variables as const but tehhowch said it was unnecessary.)